src/main/java/com/google/api/codegen/util/java/JavaTypeTable.java

```diff
@@ -116,18 +136,29 @@ public String getAndSaveNicknameFor(TypeAlias alias) {
     if (!alias.needsImport()) {
       return alias.getNickname();
     }
+
     // Derive a short name if possible
     if (imports.containsKey(alias.getFullName())) {
       // Short name already there.
       return imports.get(alias.getFullName());
     }
+    if (staticImports.containsKey(alias.getFullName())) {
+      // Short name already there.
+      return staticImports.get(alias.getFullName());
+    }
     if (imports.containsValue(alias.getNickname())
-        || !alias.getFullName().startsWith(JAVA_LANG_TYPE_PREFIX)
-            && isImplicitImport(alias.getNickname())) {
+        || staticImports.containsValue(alias.getNickname())
+        || (!alias.getFullName().startsWith(JAVA_LANG_TYPE_PREFIX)
+            && isImplicitImport(alias.getNickname()))) {
       // Short name clashes, use long name.
       return alias.getFullName();
     }
-    imports.put(alias.getFullName(), alias.getNickname());
+
+    if (alias.getParentFullName() == null) {
+      imports.put(alias.getFullName(), alias.getNickname());
+    } else {
+      staticImports.put(alias.getFullName(), alias.getNickname());
+    }
```

Comment:
I think I would prefer to put more structure around the imports instead of creating two import lists. The two import lists is forcing a bunch of duplicated logic. I think it would also be nice to not force every language to support the notion of static imports - it might only be a Java thing.